src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,143 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+
+/**
+ * Class for parsing user commands.
+ */
+public class Parser {
+    private static final String TODO_COMMAND = "todo";
+    private static final String DEADLINE_COMMAND = "deadline";
+    private static final String EVENT_COMMAND = "event";
+    private static final String LIST_COMMAND = "list";
+    private static final String DONE_COMMAND = "done";
+    private static final String FIND_COMMAND = "find";
+    private static final String DELETE_COMMAND = "delete";
+    private static final String EXIT_COMMAND = "bye";
+    private static final DateTimeFormatter dateTimeFormatter =
+            DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+
+    /**
+     * Parse user command as a String and returns the corresponding UserCommandType.
+     * @param command String of user command
+     * @return UserCommandType
+     * @throws InvalidCommandException if user command is not recognised
+     */
+    public static UserCommandType parseUserCommand(String command) throws InvalidCommandException {
+        if (command.equals(LIST_COMMAND)) {
```

Comment:
This works perfectly fine but can consider using the enum.valueOf method provided by java. In that way you don't have to declare the constants above and can use switch case instead of multiple if else statements when parsing the command. You also wont have to call the .equals string comparison method in every if clause.

Reply:
Would have to name the enum cases to be the same as the command, which I would prefer not doing